### ISSUE TITLE
Fatal Exception: java.util.concurrent.TimeoutException on close

### ISSUE THREAD

**Issue by srahimicsod** [objectbox/objectbox-java#610]
I get the following error in crashlytics , and have been unable to reproduce it on local machine in debug mode.  It seems like the app is shutting down or getting garbage collected or something like that and the boxstore.close method is not able to close within 10 seconds. Whats a good solution to this ?

Fatal Exception: java.util.concurrent.TimeoutException
io.objectbox.BoxStore.finalize() timed out after 10 seconds

o.objectbox.BoxStore.close (SourceFile:415)
io.objectbox.BoxStore.finalize (SourceFile:286)
java.lang.Daemons$FinalizerDaemon.doFinalize (Daemons.java:250)
java.lang.Daemons$FinalizerDaemon.runInternal (Daemons.java:237)
java.lang.Daemons$Daemon.run (Daemons.java:103)
java.lang.Thread.run (Thread.java:764)

**Comment by greenrobot-team** [comment]
First of all, in most cases you should not be calling `BoxStore.close()`. Just keep the `BoxStore` open for the lifetime of the app.

This exception means you are likely doing too much work while your app is in the background without having a wake lock. When the device goes to sleep for a short period, this exception might occur. Have a look at [this Stack Overflow post](https://stackoverflow.com/a/28708351/9187282). There are more resources online about this issue.

Not an ObjectBox specific problem, so nothing to do here besides performance improvements.
-ut

**Comment by srahimicsod** [comment]
We found out that the box is failing to initialize itself and raises a "Cannot change the following flags for Property" exception.(We changed a field from non-nullable to nullable) Once we get that exception , we are trying to kill the old db and re-initialize it. Since we do not have a handle on the database we simply call the static Objectbox.DeleteAllFiles and thats where we get into sort of a deadlock(Not sure how, but we see the finalizerDemon(GC??) kicking in and trying to finalize and close the objectbox since we had a deadlock(something lingering out without a good cleanup??) and that fails with a timeout(10 seconds)... we are not closing the database anywhere in our code.  

So , questions here : 
1) is there a way to tell objectbox to recreate the schema when its changed? (We went from non-nullable to a nullable and the DB is not even able to be created. can we simply tell objectbox to not throw an exception and simply blow away current data and recreate the new schema?)

2) If that is not possible, how can we safely delete old files and not get into any sort of deadlocks?
Here is the code that gets deadlocked :

    private fun getBoxStore(applicationContext: Context, dbDirectory: File): BoxStore {
        return try {
            buildBoxStore(applicationContext, dbDirectory)
        } catch (ex: io.objectbox.exception.DbException) {
            println("Database failure...probably some bad field? ${ex.localizedMessage}")
            BoxStore.deleteAllFiles(dbDirectory)

            // Yay! we deleted what we think is the old bad database.
            // Let's try one more time.
            buildBoxStore(applicationContext, dbDirectory)
        }
    }

    private fun buildBoxStore(context: Context, dbDirectory: File): BoxStore {
        val boxStore = MyObjectBox.builder()
                .androidContext(context)
                .directory(dbDirectory)
                .build()

        return boxStore
    }

This code runs in the onCreate of our Application object. Our code has been pushed to production unfortunately with this issue. (So we are trying to safely recover from this)
